tests/libvmifact/architecture.go

```diff
@@ -19,12 +19,51 @@
 
 package libvmifact
 
-var architecture string
+import "kubevirt.io/kubevirt/pkg/libvmi"
+
+const (
+	// required to start qemu on ARM with UEFI firmware
+	// https://github.com/kubevirt/kubevirt/pull/11366#issuecomment-1970247448
+	armMinimalBootableMemory = "128Mi"
+
+	defaultMinimalBootableMemory = "1Mi"
+)
+
+var (
+	architecture string
+	archAware    architectureAwareMemoryProvider = defaultArchMemoryProvider{}
+)
 
 func RegisterArchitecture(arch string) {
 	architecture = arch
+	if architecture == "arm64" {
+		archAware = arm64MemoryProvider{}
+	} else {
+		archAware = defaultArchMemoryProvider{}
+	}
+}
+
+type architectureAwareMemoryProvider interface {
+	withQEMUMinimumMemory() libvmi.Option
+	withRequestedMemory() libvmi.Option
+}
+
+type arm64MemoryProvider struct{}
```

Comment:
It can be also `arm32`? If yes, then let's simplify it and name it as `armMemoryProvider`?

Reply:
The existing code only checks for "arm64". nothing else